hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/TokenTupleUtils.java

```diff
@@ -150,25 +171,163 @@ private static List<Tuple> royaltyFeesTupleListFor(@NonNull final Token token) {
+     * @param token the token to get the fungible token info for
+     * @return Tuple encoding of the FungibleTokenInfo
+     */
+    @NonNull
+    public static Tuple fungibleTokenInfoTupleFor(@NonNull final Token token, @NonNull final String ledgerId) {
+        return Tuple.of(tokenInfoTupleFor(token, ledgerId), token.decimals());
+    }
+
+    /**
+     * Returns a tuple of the {@Code NonFungibleTokenInfo} struct
+     * {@see https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L212 }
+     * @return Tuple encoding of the NonFungibleTokenInfo
+     */
+    @NonNull
+    public static Tuple nftTokenInfoTupleFor(
+            @NonNull final Token token,
+            @NonNull final Nft nft,
+            final long serialNumber,
+            @NonNull final String ledgerId) {
+
+        final var nftMetaData = nft.metadata() != null
+                ? nft.metadata().toByteArray()
+                : com.hedera.pbj.runtime.io.buffer.Bytes.EMPTY.toByteArray();
+
+        return Tuple.of(
+                tokenInfoTupleFor(token, ledgerId),
+                serialNumber,
+                headlongAddressOf(nft.ownerIdOrElse(ZERO_ACCOUNT_ID)),
+                nft.mintTimeOrElse(new Timestamp(0, 0)).seconds(),
+                nftMetaData,
+                headlongAddressOf(nft.spenderIdOrElse(ZERO_ACCOUNT_ID)));
+    }
+
+    /**
+     * Returns a tuple of the {@code TokenKey} struct
+     * {@see https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L116 }
+     * @param keyType the type of key
+     * @param key the key for creating the tuple
+     * @return Tuple encoding of the TokenKey
+     */
+    @NonNull
+    public static Tuple typedKeyTupleFor(@NonNull final BigInteger keyType, @NonNull final Key key) {
+        return Tuple.of(
+                keyType,
+                Tuple.of(
+                        false,
+                        headlongAddressOf(key.contractIDOrElse(ZERO_CONTRACT_ID)),
+                        key.ed25519OrElse(Bytes.EMPTY).toByteArray(),
+                        key.ecdsaSecp256k1OrElse(Bytes.EMPTY).toByteArray(),
+                        headlongAddressOf(key.delegatableContractIdOrElse(ZERO_CONTRACT_ID))));
+    }
+
+    /**
+     * Returns a tuple of the {@code HederaToken} struct
+     * {@see https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L136 }
+     * @param token the token for creating the hederaToken tuple
+     * @return Tuple encoding of the HederaToken
+     */
+    @NonNull
+    private static Tuple hederaTokenTupleFor(@NonNull final Token token) {
+        //
+        final var keyList = List.of(
+                typedKeyTupleFor(BigInteger.valueOf(TokenKeyType.ADMIN_KEY.value()), token.adminKeyOrElse(Key.DEFAULT)),
```

Comment:
This file owns `TokenKeyType` (it's right below) so why not add a method to that enum that returns the `BigInteger.valueOf()` of it?